controller/resource.go

```diff
@@ -375,6 +623,37 @@ func tagValue(tags []*ecs.Tag, key string) string {
 	return ""
 }
 
-func isACLNotFoundError(err error) bool {
+// IsACLNotFoundError returns true if the ACL is not found.
+func IsACLNotFoundError(err error) bool {
 	return err != nil && strings.Contains(err.Error(), "Unexpected response code: 403 (ACL not found)")
 }
+
+// serviceNameFromDescription returns the fully qualified service name from the
+// description field of a token or policy created by the ACL controller.
+func serviceNameFromDescription(d string) string {
+	// description is of the form: "<Policy|Token> for <name> service..."
+	parts := strings.Split(d, " ")
+	if len(parts) > 3 {
+		return parts[2]
+	}
+	return ""
+}
+
+// PartitionsEnabled indicates if support for partitions and namespaces is enabled.
+func PartitionsEnabled(p string) bool {
+	return p != ""
+}
+
+// uniqueNamespaces returns the set of unique namespaces from the list of Resources.
+func uniqueNamespaces(resources []Resource) []string {
```

Comment:
What do you think about returning the set of namespaces (i.e. `map[string]struct{}`)?

Reply:
I think it's a great idea. I wasn't too sure about using "sets" but then I read the Go style guide in the eng handbook yesterday and saw that this is acceptable. I have actually just removed this func since it was only used in one spot and is vestigial now given the updated implementation.